packages/mobx-state-tree/src/core/node/node-utils.ts

```diff
@@ -119,3 +62,80 @@ export function canAttachNode(value: any) {
 export function toJSON(this: IStateTreeNode) {
     return getStateTreeNode(this).snapshot
 }
+
+export function getRelativePathBetweenNodes(base: INode, target: INode): string {
+    // PRE condition target is (a child of) base!
+    if (base.root !== target.root)
+        fail(
+            `Cannot calculate relative path: objects '${base}' and '${target}' are not part of the same object tree`
+        )
+
+    const baseParts = splitJsonPath(base.path)
+    const targetParts = splitJsonPath(target.path)
+    let common = 0
+    for (; common < baseParts.length; common++) {
+        if (baseParts[common] !== targetParts[common]) break
+    }
+    // TODO: assert that no targetParts paths are "..", "." or ""!
+    return (
+        baseParts
+            .slice(common)
+            .map(_ => "..")
```

Comment:
if this is  for a perf commit, would hoist this fn so an anon fn isn't made ~ every time it runs

Reply:
Done, thanks for noticing. I suspect modern JVM do that themselves btw